Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -60,15 +60,19 @@ class _SearchLoader<UnusedA, UnusedB>: Loader<Cursor<Site>, SearchViewController
                     for site in cursor {
                         if let url = site?.url,
                                let completion = self.completionForURL(url) {
-                            self.urlBar.setAutocompleteSuggestion(completion)
+                            DispatchQueue.main.async {
+                                self.urlBar.setAutocompleteSuggestion(completion)
+                            }
                             return
                         }
                     }
 
                     // If there are no search history matches, try matching one of the Alexa top domains.
                     for domain in self.topDomains {
                         if let completion = self.completionForDomain(domain) {
-                            self.urlBar.setAutocompleteSuggestion(completion)
+                            DispatchQueue.main.async {
```

Comment:
I'm sure this is a benefit, and I think we should do it, but three suggestions:

- I find it hard to believe that linear search as done by this `completionForDomain` pattern is as responsive as it could be. Surely we can compute an index?
- Do we get better apparent responsiveness by doing this while the query is running? If we have an index of domains in memory, can we do a quick lookup on the main thread in `didSet`?
- Can we build a domain index from the user's history and do the same thing?